src/http_upload/mod_http_upload.erl

```diff
@@ -110,6 +104,27 @@ iq_handler(_From, _To = #jid{lserver = SubHost}, Acc, IQ = #iq{type = get, sub_e
     end,
     {Acc, Res}.
 
+-spec disco_iq_handler(From :: jid:jid(), To :: jid:jid(), Acc :: mongoose_acc:t(),
+                       IQ :: jlib:iq()) ->
+          {mongoose_acc:t(), jlib:iq()}.
+disco_iq_handler(_From, _To, Acc, #iq{type = set, sub_el = SubEl} = IQ) ->
+    {Acc, IQ#iq{type = error, sub_el = [SubEl, mongoose_xmpp_errors:not_allowed()]}};
```

Comment:
add some text to not_allowed. "Disco IQ with type set are not allowed", for example.